nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/bigquery/proto/ProtoUtils.java

```diff
@@ -17,50 +17,99 @@
-        DynamicMessage.Builder builder = DynamicMessage.newBuilder(descriptor);
+   public static DynamicMessage createMessage(Descriptors.Descriptor descriptor, Map<String, Object> valueMap, TableSchema tableSchema) {
+       final DynamicMessage.Builder builder = DynamicMessage.newBuilder(descriptor);
+
+       for (final Descriptors.FieldDescriptor field : descriptor.getFields()) {
+           final String name = field.getName();
+           Object value = valueMap.get(name);
+           if (value == null) {
+               continue;
+           }
+
+           switch (field.getType()) {
+           case MESSAGE:
+               if (field.isRepeated()) {
+                   Collection collection = value.getClass().isArray() ? Arrays.asList((Object[]) value) : (Collection) value;
+                   collection.forEach(act -> builder.addRepeatedField(field, createMessage(field.getMessageType(), (Map<String, Object>) act, tableSchema)));
+               } else {
+                   builder.setField(field, createMessage(field.getMessageType(), (Map<String, Object>) value, tableSchema));
+               }
+               break;
+
+           // INT64 with alias INT, SMALLINT, INTEGER, BIGINT, TINYINT, BYTEINT
+           case INT64:
+               // Integer in the bigquery table schema maps back to INT64 which is considered to be Long on Java side:
+               // https://developers.google.com/protocol-buffers/docs/proto3
+               if (value instanceof Integer) {
+                   value = Long.valueOf((Integer) value);
+               }
+
+               setField(value, field, builder);
+               break;
+
+           // FLOAT64
+           case DOUBLE:
+               if (value instanceof Float) {
+                   value = ((Float) value).doubleValue();
+               }
+               setField(value, field, builder);
+               break;
+
+           // matches NUMERIC and BIGNUMERIC types in BigQuery
+           // BQTableSchemaToProtoDescriptor.class
+           case BYTES:
+               if(value instanceof BigDecimal) {
+                   if(tableSchema.getFields(field.getIndex()).getType().equals(Type.BIGNUMERIC)) {
+                       value = BigDecimalByteStringEncoder.encodeToBigNumericByteString((BigDecimal) value);
+                   } else if (tableSchema.getFields(field.getIndex()).getType().equals(Type.NUMERIC)) {
+                       value = BigDecimalByteStringEncoder.encodeToNumericByteString((BigDecimal) value);
+                   }
+               } else if (value instanceof Long) {
+                   value = Int64Value.of((long) value).toByteString();
+               } else if (value instanceof Float) {
+                   value = FloatValue.of((float) value).toByteString();
+               } else if (value instanceof Double) {
+                   value = DoubleValue.of((double) value).toByteString();
+               }
```

Comment:
`Integer` should be added too.
Avro `long` type is accepted for `NUMERIC` field but `int` is not.

Reply:
So I tried to consider the case where we send an Avro ``int`` into a ``NUMERIC`` column for BigQuery but this is completely messing things up. I can add something like:
````java
 } else if (value instanceof Integer) {
    value = Int64Value.of((int) value).toByteString();
 }
````
But then, in the test when checking the returned value:
````java
assertEquals(firstElt.get(0).getDoubleValue(), 1.0);
````
It would tell me:
````
org.opentest4j.AssertionFailedError: expected: <2.64E-7> but was: <1.0>
````

If sending the Avro ``int`` into a ``INT64`` column for BigQuery, then it works as expected.

I'm going to push a commit containing the additional else if so that it does not completely fail in NiFi but I believe this would push wrong data into the BQ table.